riscv/processor.cc

```diff
@@ -1049,18 +1049,18 @@ void processor_t::register_base_instructions()
   #undef DECLARE_OVERLAP_INSN
 
   #define DEFINE_INSN(name) \
-    extern reg_t rv32i_##name(processor_t*, insn_t, reg_t); \
-    extern reg_t rv64i_##name(processor_t*, insn_t, reg_t); \
-    extern reg_t rv32e_##name(processor_t*, insn_t, reg_t); \
-    extern reg_t rv64e_##name(processor_t*, insn_t, reg_t); \
+    extern reg_t fast_rv32i_##name(processor_t*, insn_t, reg_t); \
+    extern reg_t fast_rv64i_##name(processor_t*, insn_t, reg_t); \
+    extern reg_t fast_rv32e_##name(processor_t*, insn_t, reg_t); \
+    extern reg_t fast_rv64e_##name(processor_t*, insn_t, reg_t); \
     if (name##_supported) { \
       register_insn((insn_desc_t) { \
         name##_match, \
         name##_mask, \
-        rv32i_##name, \
-        rv64i_##name, \
-        rv32e_##name, \
-        rv64e_##name}); \
+        fast_rv32i_##name, \
+        fast_rv64i_##name, \
+        fast_rv32e_##name, \
+        fast_rv64e_##name}); \
```

Comment:
Why are these always the fast variants?

Edit: IIUC, 3dc8e1f244fd5ed2b06efeda1449c288f2daaffa has broken logging, which is fixed in the next commit. That's not good. We want every commit to be fully working.

Reply:
Updated.

Now, this commit adds both `fast_` and `logged_` variants, but they are identical until the last commit in this series. The last commit changes the `fast/logged` variants to use different `WRITE_REG` macros. This means its safe to use the fast variants everywhere until the last commit.